### PR TITLE
Remove Repeater dep

### PR DESCRIPTION
**Pull request by maneetgoyal**
The repeater-lite code is moved to js-clients monorepo (api-client-core package) itself for easier maintenance.

Test cases passing:

image

Test Video with a prerelease version:

https://github.com/gadget-inc/js-clients/assets/25290212/ea6cd6bb-d0a0-4010-8d73-e148dd0075a3

## PR Checklist

- [x] Important or complicated code is tested
- [ ] Any user facing changes are documented in the Gadget-side changelog
- [ ] Any immediate changes are slated for release in Gadget via a generated package dependency bump
- [ ] Versions within this monorepo are matching and there's a valid upgrade path

### CODE REVIEW

**Comment by maneetgoyal** [comment]
cc @airhorns

File: packages/api-client-core/spec/repeater.spec.ts [hunk 1]
```diff
@@ -0,0 +1,1626 @@
...
+    RepeaterOverflowError,
+    DroppingBuffer,
+    FixedBuffer,
+    MAX_QUEUE_LENGTH,
+    SlidingBuffer,
+} from "../src/repeater/index.js";
+import { delayPromise } from "./helpers.js";
+
+
+describe("RepeaterBuffer", () => {
```
Comment (airhorns) [inline packages/api-client-core/spec/repeater.spec.ts:12]: Could you add a little comment here and in the source itself explaining where this code all came from just so the next person who reads it knows that there's an upstream this is copied from?

**Review by airhorns — APPROVED** [review]
LGTM after comments

File: packages/api-client-core/spec/repeater.spec.ts (hunk 1 above)
Comment (maneetgoyal) [reply packages/api-client-core/spec/repeater.spec.ts:16]: @airhorns Done

**Pull request merged**